### PR TITLE
sandbox-classic: Fix the class loader when retrieving the banner.

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
If the classloader is not explicitly provided to `Source.fromResource`, it will use the current thread's class loader, which may not have access to _banner.txt_. To avoid issues, we need to use the correct class loader.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by SamirTalwar-DA** [comment]
This change brought to you by the following flaky test run:

```
com.daml.platform.sandbox.services.identity.LedgerIdentityServicePostgresDynamicSharedPostgresIT *** ABORTED *** (1 second, 908 milliseconds)
  java.lang.NullPointerException:
  at scala.io.Source$.fromResource(Source.scala:181)
  at com.daml.platform.sandbox.banner.Banner$.show(Banner.scala:16)
  at com.daml.ledger.sandbox.SandboxServer.initializationLoggingHeader(SandboxServer.scala:76)
  at com.daml.ledger.sandbox.SandboxServer.$anonfun$acquire$6(SandboxServer.scala:67)
  at com.daml.ledger.sandbox.SandboxServer.$anonfun$acquire$6$adapted(SandboxServer.scala:65)
```

File: ledger/sandbox-common/src/main/scala/platform/sandbox/banner/Banner.scala [hunk 1]
```diff
@@ -8,16 +8,18 @@ import java.io.PrintStream
...
-          .mkString("\n")
-      )
+  private val classLoader = getClass.getClassLoader
+
+  private val resourceName = "banner.txt"
+
+  private def banner: String =
+    if (classLoader.getResource(resourceName) != null)
+      Source
+        .fromResource(resourceName, classLoader)
```
Comment (skisel-da) [inline ledger/sandbox-common/src/main/scala/platform/sandbox/banner/Banner.scala:18]: super minor thing, but it feels a bit strange to get resource just for checking for `null` and then ignoring it.

how about this?:

```
  Option(classLoader.getResourceAsStream(resource)) match {
      case Some(in) => Source.fromInputStream(in).getLines().mkString("\n")
      case None     => "Banner resource missing from classpath."
    }
```

**Comment by tudor-da** [comment]
Thanks for raising this @SamirTalwar-DA and attempting to fix it. Do you happen to know what is the culprit here? (why would the current thread ClassLoader not have a access to the resource)

**Comment by SamirTalwar-DA** [comment]
@tudor-da: I am not sure, but I think this code could potentially end up running in a thread from an Akka-governed thread pool, which might be doing something a little strange with the class loader.

In general, asking the current thread for its class loader seems like a terrible idea.

**Review by skisel-da — APPROVED** [review]

File: ledger/sandbox-common/src/main/scala/platform/sandbox/banner/Banner.scala (hunk 1 above)
Comment (skisel-da) [reply ledger/sandbox-common/src/main/scala/platform/sandbox/banner/Banner.scala]: even better :D

**Comment by SamirTalwar-DA** [comment]
Merging to stop flakes in test runs; please feel free to investigate further if you like.

**Pull request merged**